ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -81,34 +80,26 @@ public BlobIdTest(short version) throws Exception {
   }
 
   /**
-   * Tests building blobId in {@link BlobId#BLOB_ID_V2}, and verifies values for {@code flag}, {@code datacenterId},
-   * {@code accountId}, {@code containerId}, and {@code partitionId}.
-   * @throws Exception Any unexpected exception.
+   * Tests blobId construction and assert that values are as expected.
    */
   @Test
   public void testBuildBlobId() throws Exception {
-    buildBlobIdAndAssert(referenceFlag, referenceDatacenterId, referenceAccountId, referenceContainerId,
+    BlobId blobId = new BlobId(version, referenceType, referenceDatacenterId, referenceAccountId, referenceContainerId,
         referencePartitionId);
-  }
-
-  /**
-   * Tests deserialization from a composed blobId string (not from deserialization of a blobId in both version
-   * 1 and 2.
-   * @throws Exception Any unexpected exception.
-   */
-  @Test
-  public void testDeserialization() throws Exception {
-    composedBlobIdAndDeserialize(version);
+    assertEquals("Wrong blobId version", version, getVersionFromBlobString(blobId.getID()));
+    assertBlobIdFieldValues(version, blobId, referenceType, referenceDatacenterId, referenceAccountId,
+        referenceContainerId, referencePartitionId);
   }
 
   /**
    * Tests first serializing a blobId into string, and then deserializing into a blobId object from the string.
-   * The tests are performed on both blob v1 and v2.
    * @throws Exception Any unexpected exception.
    */
   @Test
   public void testSerDes() throws Exception {
-    serDesBlobId(version);
+    BlobId blobId = new BlobId(version, referenceType, referenceDatacenterId, referenceAccountId, referenceContainerId,
+        referencePartitionId);
+    deserializeBlobIdAndAssert(version, blobId.getID());
   }
 
```

Comment:
Could you add a test for hashcode, and compareto within the same version?

Reply:
true, not much to test for hashcode. What about compareTo within the same version? I think that's useful.

Reply:
Yes and I actually added intra-version comparisons in the `testComparisons()` method (and renamed it from `testInterVersionComparison()`) as you can see below.

I am also adding a few more asserts in the SerDe test in the next commit to ensure that blob after serialization and deserialization is exactly the same as the original as far as comparisons are concerned (`equals`, `compareTo` and `hashCode()` all match exactly).